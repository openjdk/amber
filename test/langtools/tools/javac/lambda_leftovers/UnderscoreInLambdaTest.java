/*
 * @test /nodynamiccopyright/
 * @bug 8173059
 * @summary Underscore for unnamed method, constructor, lambda, and catch formals
 * @compile/fail/ref=UnderscoreInLambdaTest.out -XDrawDiagnostics UnderscoreInLambdaTest.java
 */

import java.util.function.*;

public class UnderscoreInLambdaTest {
    void foo() {
        // error
        BiFunction<Integer, String, String> biss1 =
                (_, _) ->
                    String.valueOf(_); // this use will be detected and banned by the parser
        // ok implicit
        BiFunction<Integer, String, String> biss2 =
                (_, _) -> "";
        // ok explicit
        BiFunction<Integer, String, String> biss2 =
                (int _, String _) -> "";
    }
}
