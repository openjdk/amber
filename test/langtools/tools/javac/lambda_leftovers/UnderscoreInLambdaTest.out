UnderscoreInLambdaTest.java:15:36: compiler.err.underscore.not.allowed
1 error
